# Q1

In [4]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [5]:
df=pd.read_csv("D:\Data Science\Assignments\Assignment #3 Hypothesis Testing\Cutlets.csv")
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Unit A  35 non-null     float64
 1   Unit B  35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


H(0)= There is no significant difference of the diameter of the Cutlet H(1)= There is a significant difference of the diameter of the Cutlet

In [7]:
UnitA=pd.Series(df.iloc[:,0])
UnitB=pd.Series(df.iloc[:,1])
p_value=stats.ttest_ind(UnitA,UnitB)
p_value[1]

0.4722394724599501

the p_value 0.4722>0.05(alpha value) so select null hypothesis i.e.no significiant diffrence in cutlet diameter

# Q2

In [9]:
df2=pd.read_csv("D:\Data Science\Assignments\Assignment #3 Hypothesis Testing\LabTAT.csv")
df2.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [12]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Laboratory 1  120 non-null    float64
 1   Laboratory 2  120 non-null    float64
 2   Laboratory 3  120 non-null    float64
 3   Laboratory 4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.9 KB



H(0)=All samples TAT population means are same. H(1)=Atleast one sample TAT population mean is different.

In [13]:
p_value=stats.f_oneway(df2['Laboratory 1'],df2['Laboratory 2'],df2['Laboratory 3'],df2['Laboratory 4'])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

As (p-value=0) < (alpha = 0.05); Reject Null Hypothesis i.e. Atleast one sample TAT population mean is different Thus there is variance or difference in average Turn Around Time (TAT) of reports of the laboratories on their preferred list.

# Q3

In [16]:
df3=pd.read_csv("D:\Data Science\Assignments\Assignment #3 Hypothesis Testing\BuyerRatio.csv")
df3.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [17]:
# Make dimensional array
obs=np.array([[50,142,131,70],[435,1523,1356,750]])
obs

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [18]:
# Chi2 contengency independence test
from scipy.stats import chi2_contingency
chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))


H(0)=male-female buyer rations are similar across regions and are not related. H(1)=male-female buyer rations are similar across regions and are related.

p_value > alpha i.e. (0.66 > 0.05).Accept null hyphothesis

# Q4

In [20]:
df4=pd.read_csv("D:\Data Science\Assignments\Assignment #3 Hypothesis Testing\Costomer+OrderForm.csv")
df4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free


H(0):Customer order forms defective % does not varies by centre. H(1):Customer order forms defective % does varies by centre.

In [21]:
df4['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [24]:
df4['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [25]:
df4['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [26]:
df4['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [27]:
obs1=np.array([[271,267,269,280],[29,33,31,20]])
obs1

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [29]:
chi2_contingency(obs)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

p_value > alpha i.e.(0.277 < 0.05) Accept Null Hypothesis